### PR TITLE
NO-SNOW: match pom_confluent.xml with pom.xml

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -242,9 +242,6 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>2.9.1</version>
-                <configuration>
-                    <detectJavaApiLink>false</detectJavaApiLink>
-                </configuration>
```
Comment (sfc-gh-tzhang) [inline pom.xml:247]: @sfc-gh-japatel did you add this intentionally in pom.xml but not pom_confluent.xml?

File: pom_confluent.xml [hunk 2]
```diff
@@ -427,6 +427,7 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/io.confluent/kafka-connect-avro-data -->
```
Comment (sfc-gh-rcheng) [inline pom_confluent.xml:430]: thanks for adding this in

**Review by sfc-gh-rcheng — APPROVED** [review]
Overall, do we want to collapse all our pom.xml files into one? Not sure why we need multiple

File: pom_confluent.xml (hunk 2 above)
Comment (sfc-gh-xhuang) [reply pom_confluent.xml:430]: +1

File: pom.xml (hunk 1 above)
Comment (sfc-gh-japatel) [reply pom.xml:247]: It was few years ago but based on this: https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#detectJavaApiLink

I might have seen some warnings while building this repo. May be lets keep it as is or add it to confluent_pom?

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm

File: pom.xml (hunk 1 above)
Comment (sfc-gh-tzhang) [reply pom.xml:247]: I don't see any related warning, you probably added as a workaround for some old issue, like https://issues.jenkins.io/browse/JENKINS-55692?src=confmacro. I feel like we should just use the default which is ture.

File: pom_confluent.xml [hunk 3]
```diff
@@ -7,8 +7,7 @@
 
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
-                      https://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
[the remark this answers is not in the document]
Comment (sfc-gh-tzhang) [reply pom_confluent.xml:10]: i'm not sure, i just updated the format :(

**Pull request merged**